### DISCUSSION TITLE
Question about inventory size

### DISCUSSION THREAD

**Discussion by ghost** [Minecraft-AMS/Carpet-AMS-Addition#105] [Ideas]
Would it be possible to extend the inventory rows from Shulkers above 6?
Is that still compatible with vanilla clients?

Thanks 😊

**Comment by 1024-byteeeee** [comment]
largeShulkerBox are not planned to receive any updates, and it is intended to be made as a separate mod. @Kir-Antipov Would you be willing to make it a standalone project?😊

Due to some work-related reasons, There is currently not much time to update Carpet AMS Addition, so efforts will be focused on keeping it compatible with Minecraft versions and fixing bugs.

---

largeShulkerBox 规则不计划进行任何更新，并且计划将其变为一个独立的mod， @Kir-Antipov 你愿意将其独立出去吗？😊

由于工作上的一些原因，目前没有很多时间对Carpet AMS Addition进行更新，我将尽量跟进Minecraft的版本以及对bug进行修复。

**Comment by Kira-NT** [comment]
> Would it be possible to extend the inventory rows from Shulkers above 6?

Yes.

> Is that still compatible with vanilla clients?

Not really. You see, if we want to keep this modification vanilla-friendly, we need to choose a screen handler from those provided by the base game so that clients are guaranteed to be able to process them. Minecraft has lots of built-in screen handlers, including a few "generic" ones: `9x1`, `9x2`, `9x3`, `9x4`, `9x5`, and the one and only `9x6` *(the one used by double chests in the vanilla game and by large shulker boxes here)*. However, `9x6` is the largest one that Minecraft has to offer, so, once again, we cannot go beyond that if our goal is to be as compatible with vanilla clients as possible.

> Would you be willing to make it a standalone project?😊

I have the exact same issue as you do - I'm a bit overworked at the moment and can barely keep up with my own projects.
I would be more than happy to help with any code-related problems, but setting up a new project is the most anxiety-inducing task for me, so I would rather not, you know :)

**Reply by 1024-byteeeee** [reply]
@Kir-Antipov Thanks！Complicated work is a very frustrating problem. 😥

@sternschnaube I apologize for missing your request in my last response, but Kir-Antipov provided an answer. To summarize, I may create it as an independent mod when I have time, but it will not include features that are incompatible with the vanilla clients. 😉

---

谢谢！繁杂的工作是一个让人很头大的问题。

我上次回答遗漏了你的需求，在此表示抱歉，但Kir-Antipov进行了解答，综上所述我可能会在有空的时候将其独立成Mod，但不会使它具有和原版不兼容的功能。

**Reply by ghost** [reply]
Thank you both 😄
